### ISSUE TITLE
Talkback and keyboard issues.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [2php/eyes-free#428]
```
When typing in any edit box with a bluetooth keyboard, there appears to be a 
lag between the time you hit the letter and when talkback announces what 
character you typed. And when you try to move through the typed text with your 
keyboard talkback does not announce what character you are on. This has been 
happening for some time now. I am using android 4.4.2, Samsung Galaxy S5Active 
and latest beta of talkback.What steps will reproduce the problem?
1.
2.
3.

What is the expected output? What do you see instead?


What version of the product are you using? On what operating system?


Please provide any additional information below.


```

Original issue reported on code.google.com by `AriesThe...@gmail.com` on 7 Feb 2015 at 2:53

**Comment by GoogleCodeExporter** [comment]
```
Hello,

This appears to be partially addressed. I am currently testing with Google 
Talkback 4.3.1 beta 1, Android 5.0 and I can type very fast getting proper 
voice feetback as I type. Further more navigation by letters, by words and by 
lines is working great.

When holding the shift key pressed down, the text is being selected while 
navigating however Google Talkback is not accurately reporting what and when it 
has been selected or unselected.
Also experience while navigating inside edit fields is different than we are 
used to from other enviromments. e.g. hitting home or end key or their 
combination with ctrl key move to the expected text position however Google 
talkback reports whole text we have moved to i.e. it's processing these text 
navigation commands as if we were navigating by word. I think presenting 
current character is more appropriate while doing this.

Also other parts and apps can't be consistently and accurately navigated using 
keyboard only. One huge fail while using keyboard alone is inability to get 
spoken feetback when navigating in the Gmail list of conversations.
```

Original comment by `pvdee...@gmail.com` on 5 Sep 2015 at 12:06
